packages/react-router-dom/index.tsx

```diff
@@ -1241,6 +1241,28 @@ export function useBeforeUnload(
   }, [callback, capture]);
 }
 
+/**
+ * Setup a callback to be fired on the window's `pagehide` event. This is
+ * useful for saving some data to `window.localStorage` just before the page
+ * refreshes.  This event is better supported than beforeunload across browsers.
+ *
+ * Note: The `callback` argument should be a function created with
+ * `React.useCallback()`.
+ */
+function usePageHide(
```

Comment:
Not exported currently - but if this is indeed a better way to save off some data, maybe we should export it and make a note in the docs about the use-case difference between the two hooks?